shell-ui/src/platform/service/k8s.js

```diff
@@ -0,0 +1,113 @@
+//@flow
+import { useState, useEffect } from 'react';
+import { useQuery } from 'react-query';
+import axios from 'axios';
+import { AUTHENTICATED_EVENT } from '../../navbar/events';
+
+const useGetNodesCount = (useQuery: typeof useQuery, k8sUrl: string) => {
+  const navbarElement = document.querySelector('solutions-navbar');
+  const [token, setToken] = useState(null);
+  // $FlowFixMe
+  useQuery('initialToken', () => navbarElement.getIdToken(), {
+    onSucces: (idToken) => setToken(idToken),
+  });
+
+  useEffect(() => {
+    if (!navbarElement) {
+      return;
+    }
+    const onAuthenticated = (evt: Event) => {
+      // $FlowFixMe
+      setToken(evt.detail.id_token);
+    };
+    navbarElement.addEventListener(AUTHENTICATED_EVENT, onAuthenticated);
+    return () =>
+      navbarElement.removeEventListener(AUTHENTICATED_EVENT, onAuthenticated);
+  }, []);
+
+  const queryNodesResult = useQuery(getNodesCountQuery(k8sUrl, token));
+  queryNodesResult.nodesCount = queryNodesResult.data;
+  delete queryNodesResult.data;
+
+  return queryNodesResult;
+};
+
+const useGetVolumesCount = (useQuery: typeof useQuery, k8sUrl: string) => {
+  const navbarElement = document.querySelector('solutions-navbar');
+  const [token, setToken] = useState(null);
+  // $FlowFixMe
+  useQuery('initialToken', () => navbarElement.getIdToken(), {
+    onSucces: (idToken) => setToken(idToken),
+  });
+
+  useEffect(() => {
+    if (!navbarElement) {
+      return;
+    }
+    const onAuthenticated = (evt: Event) => {
+      // $FlowFixMe
+      setToken(evt.detail.id_token);
+    };
+    navbarElement.addEventListener(AUTHENTICATED_EVENT, onAuthenticated);
+    return () =>
+      navbarElement.removeEventListener(AUTHENTICATED_EVENT, onAuthenticated);
+  }, []);
+
+  const queryVolumesResult = useQuery(getVolumesCountQuery(k8sUrl, token));
+  queryVolumesResult.volumesCount = queryVolumesResult.data;
+  delete queryVolumesResult.data;
+
+  return queryVolumesResult;
+};
+
+export const getNodesCountQuery = (
+  k8sUrl: string,
+  token?: string | null,
+): typeof useQuery => {
+  if (!token) {
+    console.error('K8s API Not authenticated');
+    return;
+  }
+  return {
+    cacheKey: 'countNodes',
+    queryFn: () =>
+      fetch(`${k8sUrl}/api/v1/nodes`)
+        .then((r) => {
+          if (r.ok) {
+            return r.json();
+          }
+        })
+        .then((res) => {
+          return res.items.length;
+        }),
+    options: { refetchInterval: 10000 },
```

Comment:
I think it is missing enabled options :

```suggestion
    options: { refetchInterval: 10000, enabled: token },
```

this will only triggers the query if the token is provided.

Reply:
Right, I've checked the doc, the `enabled` is a boolean, so maybe the following one is more correct
 ```options: { refetchInterval: 10000, enabled: token ? true : false },```

Reply:
JS is automatically casting this to a boolean so  `token ? true : false` is strictly equivalent to `token`. Actually in `token ? true : false` you are also casting token as a boolean for the ternary condition

Reply:
For some reason, the tests won't execute by `enabled: token`. So I have to change by `enabled: token ? true : false`